source/opt/pass_manager.cpp

```diff
@@ -21,6 +26,15 @@ namespace opt {
 Pass::Status PassManager::Run(ir::IRContext* context) {
   auto status = Pass::Status::SuccessWithoutChange;
   for (const auto& pass : passes_) {
+    if (print_before_all_) {
+      std::vector<uint32_t> binary;
+      context->module()->ToBinary(&binary, false);
+      SpirvTools t(SPV_ENV_UNIVERSAL_1_2);
+      std::string out;
+      t.Disassemble(binary, &out, 0);
+      std::cerr << "; IR before pass " << pass->name() << "\n"
+                << out << std::endl;
+    }
     const auto one_status = pass->Run(context);
```

Comment:
So, if this is the last pass we are running, it would be useful to print the IR after it.  The most common use I'm picturing with this is when debugging a single pass:

$ spirv-opt --print-before-all --my-pass in.spv -o out.spv

I'll want the IR after my pass is done, too.

A further enhancement to this (but not needed now) is to emit the dumps to a text file so they can be analyzed with diff.

Reply:
I modeled this after LLVM's opt -print-before-all, but I agree that would be more convenient.  How about changing the option to -print-all then it would print before each pass and once after the last pass.
I'll do that.

Reply:
LLVM's print-before-all writes to stderr, which is why I did stderr.  However, I realize the API should use an output stream rather than a fixed file.  I'll make that change too.